lib/spack/spack/cmd/create.py

```diff
@@ -771,26 +804,84 @@ def get_name(args):
     return name
 
 
-def get_url(args):
-    """Get the URL to use.
+def is_git_url(url):
+    """Check if the URL is likely to be a git repository. The code doesn't attempt
+    to clone the repository!
+
+    Args:
+        url (str): The url to check
+
+    Returns:
+        bool: True if it seems to be a git repository
+    """
+
+    if url is None:
+        return False
+
+    try:
+        schema, user, host, port, path = spack.util.url.parse_git_url(url)
+    except ValueError:
+        # While /srv/git/project.git is a valid Git url,
+        # let's be pessimistic and assume that this is not
+        # what the user wants
+        return False
+    else:
+        try:
+            path = path.rsplit('/', 1)[1]
+        except (IndexError, AttributeError):
+            # path is None or doesn't contain '/'
+            return False
+
+        if path.endswith('.git'):
+            return True
 
-    Use a default URL if none is provided.
+        if (schema is None or schema == 'ssh://') and user == 'git':
+            return True
+
+        return False
+
+
+def get_url_and_git(args):
+    """Get the source and git URLs to use.
+
+    Use defaults if none is provided.
 
     Args:
         args (argparse.Namespace): The arguments given to ``spack create``
 
     Returns:
-        str: The URL of the package
+        tuple(str, Optional[str]): The source and git URLs of the package
     """
 
-    # Default URL
+    # Possible flag combinations:
+    # spack create or spack create -g -> args.git is None, args.url is None
+    # spack create <url> -> args.git = '#AUTO-GIT-URL#', args.url = <url>
+    #       -- in this case, url is checked to be git-like
+    # spack create -g <url> -> args.git = <url>, args.url is None
+    # spack create -g [some other flag that consumes argument] <url>
+    #               -> args.git is None, args.url = <url>
+    # spack create -g <git_url> <url> -> args.git = <git_url>, args.url = <url>
+
+    # Default URLs
     url = 'https://www.example.com/example-1.2.3.tar.gz'
+    git = None
+
+    # No source and no git urls were provided
+    if args.url is None and args.git is None:
+        return url, git
+
+    # Git url not set explicitly
+    if (args.git == '#AUTO-GIT-URL#' and is_git_url(args.url)) or args.git is None:
+        git = args.url
+        return url, git
```

Comment:
Would it be helpful to return the following (so the `url` is not the example?

```suggestion
        return args.url, git
```